### PR TITLE
ECDC-2725 Enable stream advanced options

### PR DESCRIPTION
**Pull request by kmurica**
### Issue

ECDC-2725

### CODE REVIEW

File: nexus_constructor/model/helpers.py [hunk 1]
```diff
@@ -14,8 +14,13 @@ def __find_item_index(list_to_look_in: List[Any], item_name: str):
...
     """
     for count, element in enumerate(list_to_look_in):
-        if element.name == item_name:
-            return count
+        try:
+            if element.name == item_name:
+                return count
+        except AttributeError:
+            continue
+    else:
```
Comment (ebadkamil) [inline nexus_constructor/model/helpers.py:22]:
```suggestion
    return None
    
```

File: nexus_constructor/stream_fields_widget.py [hunk 2]
```diff
@@ -216,13 +213,18 @@ def _set_up_f142_group_box(self):
             parent=self.show_advanced_options_button
         )
         self.f142_advanced_group_box.setLayout(QFormLayout())
+        self.f142_adc_pulse_debug_checkbox = QCheckBox()
```
Comment (ebadkamil) [inline nexus_constructor/stream_fields_widget.py:216]: Could you remind me why this new checkbox is introduced?

File: nexus_constructor/stream_fields_widget.py [hunk 3]
```diff
@@ -337,24 +339,26 @@ def _record_advanced_f142_values(self, stream: F142Stream):
         Save the advanced f142 properties to the stream data object.
         :param stream: The stream data object to be modified.
         """
-        raise NotImplementedError
+        stream.adc_pulse_debug = self.f142_adc_pulse_debug_checkbox.isChecked()
```
Comment (ebadkamil) [inline nexus_constructor/stream_fields_widget.py:342]: `adc_pulse_debug` is not in F142 schema. It is only for ev42.

https://github.com/ess-dmsc/kafka-to-nexus/blob/main/documentation/writer_module_f142_log_data.md

File: nexus_constructor/stream_fields_widget.py (hunk 2 above)
Comment (ebadkamil) [reply nexus_constructor/stream_fields_widget.py:216]: See comments below.

File: nexus_constructor/model/module.py [hunk 4]
```diff
@@ -97,20 +97,26 @@ def as_dict(self, error_collector: List[str]):
 
 
 @attr.s
-class F142Stream(EV42Stream):
+class F142Stream(StreamModule):
+    cue_interval = attr.ib(type=int, default=None, init=False)
```
Comment (ebadkamil) [inline nexus_constructor/model/module.py:101]: `init=False ` is not required probably for `cue_interval` and `chunk_size`

**Comment by ebadkamil** [comment]
One minor comment above. LGTM

File: nexus_constructor/model/module.py (hunk 4 above)
Comment (kmurica) [reply nexus_constructor/model/module.py]: True! Done

**Review by ebadkamil — APPROVED** [review]

**Pull request merged**